source/extensions/http/cache/file_system_http_cache/file_system_http_cache.cc

```diff
@@ -388,6 +405,78 @@ void FileSystemHttpCache::trackFileRemoved(uint64_t file_size) {
   stats_.size_bytes_.set(size_bytes_);
 }
 
+void FileSystemHttpCache::maybeEvict() {
+  uint64_t size_to_evict = 0;
+  uint64_t count_to_evict = 0;
+  if (config().has_max_cache_size_bytes()) {
+    // capture a value from the atomic because we're going to use it twice and don't want our
+    // value to change in between.
+    uint64_t seen_size = size_bytes_;
+    size_to_evict = seen_size > config().max_cache_size_bytes().value()
+                        ? seen_size - config().max_cache_size_bytes().value()
+                        : 0;
+  }
+  if (config().has_max_cache_entry_count()) {
+    // capture a value from the atomic because we're going to use it twice and don't want our
+    // value to change in between.
+    uint64_t seen_count = size_count_;
+    count_to_evict = seen_count > config().max_cache_entry_count().value()
+                         ? seen_count - config().max_cache_entry_count().value()
+                         : 0;
+  }
+  if (size_to_evict == 0 && count_to_evict == 0) {
+    return;
```

Comment:
can we factor lines 409-429 into a needsCleanup() method?

Then it might be easier to run that prior to signalling, and would also be easy to test.

Reply:
Done-ish. needsEviction, because Cleanup means too many other things, and I ended up with two of them because here we need to capture the size and in the other place we don't (and doing so anyway would be awkward and confusing).